src/raw_client.rs

```diff
@@ -467,14 +485,66 @@ impl RawClient<ElectrumProxyStream> {
         Ok(stream.into())
     }
 
-    #[cfg(any(
-        feature = "use-openssl",
-        feature = "use-rustls",
-        feature = "use-rustls-ring"
+    #[cfg(all(
+        any(
+            feature = "default",
+            feature = "use-rustls",
+            feature = "use-rustls-ring"
+        ),
+        not(feature = "use-openssl")
     ))]
     /// Creates a new TLS client that connects to `target_addr` using `proxy_addr` as a socks proxy
     /// server. The DNS resolution of `target_addr`, if required, is done through the proxy. This
     /// allows to specify, for instance, `.onion` addresses.
+    pub fn new_proxy_ssl<T: ToTargetAddr>(
```

Comment:
Is this adding a new feature where you can use a proxy with rustls? I'm a little confused why this is being added in this PR.

Reply:
I did it because `CryptoProvider` is specific to `rustls`, so I had to split it into `rustls` and `openssl` feature-gated fns.

Reply:
Hmmm I think before doing this PR we should do some refactoring. There is way too much feature gating logic to me.

1. We shouldn't have a method like `new_proxy_ssl`, just have a single method that starts an electrum TLS connection given a `TcpStream`. Of course then we have a convenience method to do the initial connection.  
2. Do we even need openssl as a dependency at all? If someone wants to use socks5 then can depend on openssl themelves or use one of the many pure rust socks5 client libraries like [socks5-client](https://docs.rs/socks5-client/latest/socks5_client/). We can have an example showing how to do tor. IMO it'd be better to try and use `socks5-client` rather than openssl. For TLS we can just be opinionated and use rustls.

This seems like something you would want to do anyway @oleonardolima for arti?

I'll turn this into an issue that I think we should tackle first.

Reply:
Yes, I do agree with both (1) and (2), there are also issues with other features such s `proxy` previously mentioned by tnull, refer to #91, which is basically mandatory and we can't use the client with `--no-default-features`.

Indeed, with `arti-client` I do need the client to receive a data stream (e.g. `TcpStream`, `arti-client::DataStream`), and was planning on such refactoring. It also requires being async, which detangling all existing features will make it easy to have an async version.

Thanks for opening the issue, I'll move this PR back to draft for now.

> I also commented on the issue, for anyone else who didn't look at this PR.